File: python/tests/test_CDPExtensionInjector.py
# MODCDP_TRANSLATE_TEST: KEEP THIS TEST FILE TRANSLATED ACROSS TYPESCRIPT, PYTHON, AND GO.
# All test cases, descriptions, covered edge cases, and setup should be kept perfectly 1:1 in sync between:
# - ./js/test/test.CDPExtensionInjector.ts
# - ./go/modcdp/injector/CDPExtensionInjector_test.go
# NO MOCKING, NO MONKEY PATCHING, NO SIMULATING, NO FAKING, NO SKIPPING ALLOWED.
# USE REAL USER-FACING CODE PATHS WITH REAL BROWSERS, REAL CLASSES, REAL URLS, etc. Hard fail if keys or other env requirements are missing.
from __future__ import annotations

import unittest
from pathlib import Path

from modcdp.injector.CDPExtensionInjector import CDPExtensionInjector


class CDPExtensionInjectorTests(unittest.TestCase):
    def test_cdpextensioninjector_prepares_the_default_packaged_extension_zip(self) -> None:
        injector = CDPExtensionInjector()
        try:
            injector.prepare()
            unpacked_extension_path = injector.unpacked_extension_path
            if not isinstance(unpacked_extension_path, str):
                self.fail(f"unpacked_extension_path = {unpacked_extension_path!r}")
            self.assertIn("modcdp-extension-", unpacked_extension_path)
            self.assertTrue((Path(unpacked_extension_path) / "manifest.json").exists())
        finally:
            injector.close()


if __name__ == "__main__":
    unittest.main()

File: js/test/test.CDPExtensionInjector.ts
// MODCDP_TRANSLATE_TEST: KEEP THIS TEST FILE TRANSLATED ACROSS TYPESCRIPT, PYTHON, AND GO.
// All test cases, descriptions, covered edge cases, and setup should be kept perfectly 1:1 in sync between:
// - ./python/tests/test_CDPExtensionInjector.py
// - ./go/modcdp/injector/CDPExtensionInjector_test.go
// NO MOCKING, NO MONKEY PATCHING, NO SIMULATING, NO FAKING, NO SKIPPING ALLOWED.
// USE REAL USER-FACING CODE PATHS WITH REAL BROWSERS, REAL CLASSES, REAL URLS, etc. Hard fail if keys or other env requirements are missing.
import assert from "node:assert/strict";
import { existsSync } from "node:fs";
import path from "node:path";
import { test } from "vitest";

import { CDPExtensionInjector } from "../src/injector/CDPExtensionInjector.js";

test("CDPExtensionInjector prepares the default packaged extension zip", async () => {
  const injector = new CDPExtensionInjector();

  try {
    await injector.prepare();
    const unpacked_extension_path = (injector as unknown as { unpacked_extension_path?: string | null })
      .unpacked_extension_path;
    assert.equal(typeof unpacked_extension_path, "string");
    assert.match(unpacked_extension_path!, /modcdp-extension-/);
    assert.equal(existsSync(path.join(unpacked_extension_path!, "manifest.json")), true);
  } finally {
    await injector.close();
  }
});
